src/core/searcher.rs

```diff
@@ -83,7 +84,7 @@ impl Searcher {
     ///
     /// The searcher uses the segment ordinal to route the
     /// request to the right `Segment`.
-    pub fn doc(&self, doc_address: DocAddress) -> crate::Result<Document> {
+    pub fn doc<D: DocumentAccess>(&self, doc_address: DocAddress) -> crate::Result<D> {
```

Comment:
Should this generic be moved to the searcher so we can have a default?

Reply:
If I remember right, adding the generic to the searcher itself caused a whole world of pain with us referencing it everywhere and using it in places like the segment readers